core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -434,6 +438,46 @@ class TransactionsTest extends KafkaServerTestHarness {
     }
   }
 
+  @Test
+  def testMultipleMarkersOneLeader(): Unit = {
```

Comment:
Note, this doesn't reproduce the original problem. Leaving it out here in case @hachikuji or someone else has an idea about when multiple invocations of the response callback can happen for a single append to the log.